src/compiler/types.ts

```diff
@@ -1293,13 +1293,6 @@ export type HasIllegalTypeParameters =
     | SetAccessorDeclaration
     | GetAccessorDeclaration;
 
-export type HasTypeArguments =
-    | CallExpression
-    | NewExpression
-    | TaggedTemplateExpression
-    | JsxOpeningElement
-    | JsxSelfClosingElement;

```

Comment:
I removed this along with the `hasTypeArguments` guard, but I think this type is public so I could restore it if this might cause breakage somewhere (even if the internal `hasTypeArguments` remains purged).

If you'd like me to bring this back, perhaps `NodeWithTypeArguments` should be added to the union?